database-plugins/src/main/java/io/cdap/plugin/DBUtils.java

```diff
@@ -284,15 +280,7 @@ public static Object transformValue(int sqlType, int precision, int scale,
         case Types.NUMERIC:
         case Types.DECIMAL:
           BigDecimal decimal = (BigDecimal) original;
-          if (scale != 0) {
-            // if there are digits after the point, use double types
-            return decimal.doubleValue();
-          } else if (precision > 9) {
-            // with 10 digits we can represent 2^32 and LONG is required
-            return decimal.longValue();
-          } else {
-            return decimal.intValue();
-          }
+          return decimal.unscaledValue().toByteArray();
```

Comment:
Please move this logic here: 
https://github.com/cdapio/hydrator-plugins/blob/develop/database-plugins/src/main/java/io/cdap/plugin/DBRecord.java#L137